File: japiwznm/WznmQRtjList.java
/**
	* \file WznmQRtjList.h
	* Java API code for record of table TblWznmQRtjList
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

package apiwznm;

import org.w3c.dom.*;
import sbecore.*;

public class WznmQRtjList {
	
	public WznmQRtjList(
				int jnum
				, String sref
				, String stubRefWznmMApp
				, String stubSupRefWznmMRtjob
				, String stubRefWznmMJob
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.stubRefWznmMApp = stubRefWznmMApp;
		this.stubSupRefWznmMRtjob = stubSupRefWznmMRtjob;
		this.stubRefWznmMJob = stubRefWznmMJob;
	};

	public int jnum;
	public String sref;
	public String stubRefWznmMApp;
	public String stubSupRefWznmMRtjob;
	public String stubRefWznmMJob;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WznmQRtjList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			stubRefWznmMApp = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMApp", "app", null, 0);
			stubSupRefWznmMRtjob = Xmlio.extractStringUclc(doc, basexpath, "stubSupRefWznmMRtjob", "sup", null, 0);
			stubRefWznmMJob = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMJob", "job", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apiwznm/WznmQRtjList.cpp
/**
	* \file WznmQRtjList.cpp
	* API code for table TblWznmQRtjList (implementation)
	* \copyright (C) 2016-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

#include "WznmQRtjList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WznmQRtjList
 ******************************************************************************/

WznmQRtjList::WznmQRtjList(
			const uint jnum
			, const string sref
			, const string stubRefWznmMApp
			, const string stubSupRefWznmMRtjob
			, const string stubRefWznmMJob
		) {
	this->jnum = jnum;
	this->sref = sref;
	this->stubRefWznmMApp = stubRefWznmMApp;
	this->stubSupRefWznmMRtjob = stubSupRefWznmMRtjob;
	this->stubRefWznmMJob = stubRefWznmMJob;
};

bool WznmQRtjList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WznmQRtjList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "stubRefWznmMApp", "app", stubRefWznmMApp);
		extractStringUclc(docctx, basexpath, "stubSupRefWznmMRtjob", "sup", stubSupRefWznmMRtjob);
		extractStringUclc(docctx, basexpath, "stubRefWznmMJob", "job", stubRefWznmMJob);
	};

	return basefound;
};

/******************************************************************************
 class ListWznmQRtjList
 ******************************************************************************/

ListWznmQRtjList::ListWznmQRtjList() {
};

ListWznmQRtjList::ListWznmQRtjList(
			const ListWznmQRtjList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQRtjList(*(src.nodes[i]));
};

ListWznmQRtjList& ListWznmQRtjList::operator=(
			const ListWznmQRtjList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQRtjList(*(src.nodes[i]));

	return *this;
};

ListWznmQRtjList::~ListWznmQRtjList() {
	clear();
};

void ListWznmQRtjList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWznmQRtjList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WznmQRtjList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWznmQRtjList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WznmQRtjList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WznmQRtjList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WznmQRtjList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

